docs/topics/Rate_Limits.md

```diff
@@ -37,11 +37,11 @@ In the case that a rate limit is exceeded, the API will return a HTTP 429 respon
 
 ###### Rate Limit Response Structure
 
-| Field       | Type    | Description                                                           |
-| ----------- | ------- | --------------------------------------------------------------------- |
-| message     | string  | A message saying you are being rate limited.                          |
-| retry_after | integer | The number of milliseconds to wait before submitting another request. |
-| global      | boolean | A value indicating if you are being globally rate limited or not      |
+| Field       | Type             | Description                                                      |
+|-------------|------------------|------------------------------------------------------------------|
+| message     | string           | A message saying you are being rate limited.                     |
+| retry_after | integer or float | The number of seconds to wait before submitting another request. |
```

Comment:
Per [JSON specifications](https://www.json.org/json-en.html), you only have a numeric type, which may be treated as an integer-style value or a floating point-style value, so is there any real benefit in documenting that it can be an integer _and_ a float, if standards compliant JSON parsers should handle both cases anyway?

Reply:
While some JSON parsers themselves might be able to automatically detect the numeric type, it is still incredibly helpful for library/bot devs to know which fields are integers and which are floats without having to manually inspect every possible response body. This is especially important for more strictly typed languages. The JSON parser is only a small part of the entire library/bot.

Reply:
I guess my point was worded poorly; my mistake.

I was more getting at the point that perhaps this should just be documented as being a numeric value with a fractional part, not an `or`. Especially in static typed languages where integral types are treated as primitives, it probably is a bit silly to suggest that you have to implement a case that handles an integer type in this field and a case that handles a floating point type in this field. Especially when `1.0` would have the same meaning here that `1` would.

I guess I was trying to say that this seems like a bit of an irrelevant piece of information to give, as you would assume that the developer would know that you can cast between an integer and a floating point type in the language of their choice given no fractional component on the value.

Reply:
Yeah this is weird. If it can have an important fractional part, it should be `float` or `number`.

Reply:
```suggestion
| retry_after | float | The number of seconds to wait before submitting another request. |
```

In most (all?) cases it's a number with decimals.

Reply:
@tpcstld that is much more idiomatic and obvious. Saying `integer` or `float` could be seen as implying they are two different cases.

For consistency, it would probably be more sensible if the API just gave a specific number of decimal places of precision, as well, but I don't see that changing any time soon.